### PR TITLE
Deprecate legacy data source embedder config

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

Following the changes in https://github.com/dust-tt/dust/pull/5284, all data sources have been backfilled to use the new embedder config structure. This PR deprecates the legacy embedder config and includes the following updates:
- Updates the core logic to use the new embedder config.
- Refactor  the core types to remove deprecated fields.
- Adjusts the types and API definition to exclude the legacy format.

## Risk

Worst case it breaks data sources in core. Safe to rollback but it will required to backfill the missing information.

## Deploy Plan

⚠️ First need to deploy `core` and then deploy `front`.

### CODE REVIEW

File: core/bin/dust_api.rs [hunk 1]
```diff
@@ -1182,8 +1182,8 @@ async fn data_sources_tokenize(
                 None,
             ),
             Some(ds) => {
-                let config = ds.config().clone();
-                let llm = provider(config.provider_id).llm(config.model_id);
+                let embedder_config = ds.default_embedder_config().clone();
+                let llm = provider(embedder_config.provider_id).llm(embedder_config.model_id);
```
Comment (spolu) [inline core/bin/dust_api.rs:1186]: Not related to this PR but this is off.

Should be `embedder` not `llm`. Should be equivalent for OpenAI.

File: core/src/data_sources/data_source.rs [hunk 2]
```diff
@@ -537,6 +531,10 @@ impl DataSource {
         &self.config
     }
 
+    pub fn default_embedder_config(&self) -> &EmbedderConfig {
```
Comment (spolu) [inline core/src/data_sources/data_source.rs:534]: Can we drop the `default` ?

File: core/bin/dust_api.rs (hunk 1 above)
Comment (flvndvd) [reply core/bin/dust_api.rs:1186]: Track here: https://github.com/dust-tt/dust/issues/5296.

File: core/src/data_sources/data_source.rs (hunk 2 above)
Comment (spolu) [reply core/src/data_sources/data_source.rs:534]: yes

Comment (flvndvd) [reply core/src/data_sources/data_source.rs]: Updated.

Comment (spolu) [reply core/src/data_sources/data_source.rs]: and we'll have `shadow_write_embedder_config` in the future. It's just that `default` does not carry any real meaning here.

Comment (flvndvd) [reply core/src/data_sources/data_source.rs]: Agree, it was mainly to avoid the confusion between the field name the function name.

**Review by spolu — APPROVED** [review]

**Pull request merged**